devdocs/godmode.md

```diff
@@ -0,0 +1,69 @@
+#GodMode User Notes
+
+##What is GodMode?
+
+Typically browser tests involve comparing the source of the webpage to an existing *expected* source. This serves to verify the DOM structure (HTML) of the page. However, creating new browser tests and updating outdated ones require considerable effort on the developer's part. In addition, ensuring that the generated *expected* source code works across computers, browsers, user accountsa, and execution times is harder still. To overcome this problem, GodMode provides a simple way to create and update *expected* source files for browser tests and ensure the necessary cross-compatibility.
+
+
+##How does GodMode work?
+
+The essential idea is to reverse the process of testing, whereby the actual source of the webpage during the test is used to overwrite the *expected* source for the test, as opposed to the usual way of using the *expected* source as the reference and checking the actual source of the webpage. In actual implementation we check if the test already passes and only if it fails do we update the *expected* source. Thus running GodMode on passing tests poses no danger.
+
+
+##How do we use GodMode?
+
+GodMode can be activated in two different ways. 
+
+1. If we want to execute arbitrary tests using GodMode, then update BaseUiTestCase class and set `enableGodMode = true` at the top of the class implementation. Now all test runs would have GodMode enabled. Please remember to set it back to false when done.
+
+2. If we want to run a particular test suite using GodMode, then go to `Run -> Run Configurations` and update the appropriate one with the `-Dgodmode=true` VM argument. Please remember to remove the argument before committing the changes
+
```

Comment:
The idea was to simplify the usage of Godmode. It makes it easier for new devs to use, and for reviewers to ensure godmode is not left on by mistake. We don't really need the capability to update just one test. I agree it is convenient, but those who really want it would figure it out :-P